src/linear_algebra/finsupp.lean

```diff
@@ -802,6 +802,13 @@ begin
   exact hN i hi (hg _),
 end
 
+/-- `submodule.exists_finset_of_mem_supr` as an `iff` -/
+lemma submodule.mem_supr_iff_exists_finset
+  {ι : Sort*} {p : ι → submodule R M} {m : M} :
+  (m ∈ ⨆ i, p i) ↔ ∃ s : finset ι, m ∈ ⨆ i ∈ s, p i :=
+⟨submodule.exists_finset_of_mem_supr p,
+ λ h, let ⟨_, hs⟩ := h in supr_le_supr (λ i, (supr_const_le : _ ≤ p i)) hs⟩
```

Comment:
Does
```suggestion
 λ ⟨_, hs⟩, supr_le_supr (λ i, (supr_const_le : _ ≤ p i)) hs⟩
```
not work?

Reply:
Good catch, I was thinking about data fields where that approach leaves a mess, but that's not relevant here.